cmd/main_test.go

```diff
@@ -301,3 +307,122 @@ var _ = DescribeTable("TestRunAccessCheck", func(user string, namespace string,
 		"patch",
 		false),
 )
+
+var _ = DescribeTable("GetWorkspacesWithAccess querying for workspaces with access", func(gv string, allNamespaces []k8sapi.Namespace, expectedWorkspaces []crt.Workspace) {
+	e := echo.New()
+	c := e.NewContext(nil, nil)
+	Context("When workspace test-tenant's namespaces has all the necessary permissions", func() {
+		BeforeEach(func() {
+			gv = "v1alpha1"
+			namespaceNames := []string{"test-tenant", "test-tenant-2"}
+			for _, name := range namespaceNames {
+				ns, err := createNamespace(k8sClient, name)
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace %s: %v", name, err))
+				allNamespaces = append(allNamespaces, ns)
+				createRole(k8sClient, name, "namespace-access", []string{"create", "list", "watch", "delete"})
+				createRoleBinding(k8sClient, "namespace-access-user-binding", name, "user1@konflux.dev", "namespace-access")
```

Comment:
Those two lines should result in an error, as a resource with this name ( "namespace-access-user-binding", "namespace-access") already exists (as you are creating them for two namespaces). if it did not error, check out why. 

after you see why, you can use an index to create it in the loop.